auth0/resource_auth0_connection.go

```diff
@@ -367,6 +367,48 @@ func newConnection() *schema.Resource {
 							Optional: true,
 							Elem:     &schema.Schema{Type: schema.TypeString},
 						},
+
+						// OIDC options
+						"type": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "",
+						},
+						"scope": {
```

Comment:
Can we use `scopes` for this? With the SDK all the oauth2/openid connections have a SetScopes method to deal with this easily. You could have a look at the google connection how that does it.

Reply:
Also see the `expandConnectionOptionsScopes` I believe its relevant to this.

Reply:
Thanks for your review. First things first, this is the first time I touch Go so while I am able to reproduce stuff that already exists, it's quite challenging for me to come up with new stuff!

I saw that setScopes function but [it's not present for the OIDC connection in the SDK](https://github.com/go-auth0/auth0/blob/9fd4a3c6d5ac94db0161db19ee5d16b0e04216ca/management/connection.go#L481), it seems to be the only one connection in this case.

Reply:
No worries at all! I'm happy to help. If you are having any trouble just let me know.

You are right, I must have missed a bunch `SetScopes` in go-auth0/auth0. I can do that from tomorrow/weekend or if you're in a hurry, feel free to patch it yourself. You can use the same one as Salesforce above OIDC.

You can use the `GoogleOAuth2` as your guiding principle. Also #221 and #222 are great examples, very similar to your PR.

Reply:
I was about to do it but am not sure how to test it and have the `management.gen.go` generated (I suppose it is)... So I prefer to leave it to you.